[glib] Build error

<!-- ⚠ Please use the generated issue template instead to report a port build failure. ⚠ -->
<!-- ⁉ You can find its location at the end of the console output. ⁉ -->

**Host Environment**

- OS: Windows 11
- Compiler: MSVC 17

**To Reproduce**

Steps to reproduce the behavior:

`./vcpkg install glib:x64-windows`

**Failure logs**

C:\Users\15819>vcpkg install glib:x64-windows
Computing installation plan...
The following packages will be built and installed:
    glib[core]:x64-windows -> 2.74.0
Detecting compiler hash for triplet "x64-windows"...
-- Using HTTP(S)_PROXY in environment variables.
Restored 0 packages from C:\Users\15819\AppData\Local\vcpkg\archives in 126.7 us. Use --debug to see more details.
Installing 1/1 glib:x64-windows...
Building glib[core]:x64-windows...
-- Using cached glib-2.74.0.tar.xz.
-- Cleaning sources at F:/vcpkg/buildtrees/glib/src/glib-2-22354247da.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source F:/vcpkg/downloads/glib-2.74.0.tar.xz
-- Applying patch use-libiconv-on-windows.patch
-- Applying patch libintl.patch
-- Using source at F:/vcpkg/buildtrees/glib/src/glib-2-22354247da.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x64-windows
-- Configuring x64-windows-dbg
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at F:/vcpkg/downloads/tools/msys2/9a1ec3f33446b195
-- Configuring x64-windows-dbg done
-- Configuring x64-windows-rel
-- Configuring x64-windows-rel done
-- Package x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: "D:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" install -v
    Working Directory: F:/vcpkg/buildtrees/glib/x64-windows-dbg
    Error code: 1
    See logs for more information:
      F:\vcpkg\buildtrees\glib\package-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_meson.cmake:33 (vcpkg_execute_required_process)
  ports/glib/portfile.cmake:47 (vcpkg_install_meson)
  scripts/ports.cmake:147 (include)


error: building glib:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+glib
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[glib]+Build+error
Include '[glib] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-06-17-9268e366206712e38102b28dbd1617697a99ff2e
    vcpkg-scripts version: 934a99dc1 2022-10-21 (34 hours ago)

You can also use the prefilled template from F:\vcpkg\installed\vcpkg\issue_body.md.

**Additional context**

Methods I tried:

 - Reboot
 - Using PowerShell under Administrator
 - ```git pull``` & ```vcpkg update```

> > Will it be released soon? Or can I use the fixed version first before officially released?
> 
> glib 2.74.1 was release 4 days ago... Can you test if #27554 solves the issue?

I have run this again, it doesn't work at all. My command and output:
```
PS F:\vcpkg> git pull
Already up to date.
PS F:\vcpkg> vcpkg update
Using local portfile versions. To update the local portfiles, use `git pull`.
The following packages differ from their port versions:
    boost-modular-build-helper:x64-windows 1.80.0#2 -> 1.80.0#3
    ffmpeg:x64-windows               4.4.1#21 -> 4.4.1#22
    fontconfig:x86-windows           2.14.0#4 -> 2.14.1
    libmysql:x64-windows             8.0.20#8 -> 8.0.20#9
    qt5-base:x64-windows             5.15.6#1 -> 5.15.6#2
    vcpkg-tool-nodejs:x64-windows    14.17.4 -> 16.18.0
    zlib:x64-windows                 1.2.12#2 -> 1.2.13

To update these packages and all dependencies, run
    .\vcpkg upgrade

To only remove outdated packages, run
    .\vcpkg remove --outdated
PS F:\vcpkg> vcpkg install glib glib:x64-windows
Computing installation plan...
The following packages will be built and installed:
    glib[core]:x64-windows -> 2.74.0
    glib[core]:x86-windows -> 2.74.0
Detecting compiler hash for triplet x64-windows...
-- Using HTTP(S)_PROXY in environment variables.
Detecting compiler hash for triplet x86-windows...
Restored 0 package(s) from C:\Users\15819\AppData\Local\vcpkg\archives in 126.5 us. Use --debug to see more details.
Installing 1/2 glib:x64-windows...
Building glib[core]:x64-windows...
-- Downloading https://download.gnome.org/sources/glib/2.74/glib-2.74.0.tar.xz -> glib-2.74.0.tar.xz...
-- Cleaning sources at F:/vcpkg/buildtrees/glib/src/glib-2-22354247da.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source F:/vcpkg/downloads/glib-2.74.0.tar.xz
-- Applying patch use-libiconv-on-windows.patch
-- Applying patch libintl.patch
-- Using source at F:/vcpkg/buildtrees/glib/src/glib-2-22354247da.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x64-windows
-- Configuring x64-windows-dbg
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at F:/vcpkg/downloads/tools/msys2/9a1ec3f33446b195
-- Configuring x64-windows-dbg done
-- Configuring x64-windows-rel
-- Configuring x64-windows-rel done
-- Package x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: "D:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" install -v
    Working Directory: F:/vcpkg/buildtrees/glib/x64-windows-dbg
    Error code: 1
    See logs for more information:
      F:\vcpkg\buildtrees\glib\package-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_meson.cmake:33 (vcpkg_execute_required_process)
  ports/glib/portfile.cmake:47 (vcpkg_install_meson)
  scripts/ports.cmake:147 (include)


error: building glib:x64-windows failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+glib
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[glib]+Build+error
Include '[glib] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-10-17-3247920fbdd47d08f36cbd480addd9890d3c2435
    vcpkg-scripts version: 253704407 2022-10-28 (32 hours ago)


Please use the prefilled template from F:\vcpkg\installed\vcpkg\issue_body.md when reporting your issue.
```

It seems that vcpkg is still using glib 2.74.0

--- a/ports/glib/libintl.patch
+++ b/ports/glib/libintl.patch
@@ -2,46 +2,16 @@ diff --git a/meson.build b/meson.build
 index f44fa2d4e..d465253af 100644
 --- a/meson.build
 +++ b/meson.build
-@@ -2090,37 +2090,10 @@ libz_dep = dependency('zlib')
+@@ -2090,9 +2090,9 @@ libz_dep = dependency('zlib')
  # proxy-libintl subproject.
  # FIXME: glib-gettext.m4 has much more checks to detect broken/uncompatible
  # implementations. This could be extended if issues are found in some platforms.
 -libintl_deps = []
 -libintl = dependency('intl', required: false)
 -if libintl.found()
--  # libintl supports different threading APIs, which may not
--  # require additional flags, but it defaults to using pthreads if
--  # found. Meson's "threads" dependency does not allow you to
--  # prefer pthreads. We may not be using pthreads for glib itself
--  # either so just link the library to satisfy libintl rather than
--  # also defining the macros with the -pthread flag.
--  #
--  # Meson's builtin dependency lookup as of 0.60.0 doesn't check for
--  # pthread, so we do this manually here.
--  if cc.has_function('ngettext', dependencies : libintl)
--    libintl_deps += [libintl]
--  else
--    libintl_pthread = cc.find_library('pthread', required : false)
--    if libintl_pthread.found() and cc.has_function('ngettext', dependencies : [libintl, libintl_pthread])
--      libintl_deps += [libintl, libintl_pthread]
--    else
--      libintl = disabler()
--    endif
--  endif
--endif
--
--if libintl.found()
--  have_bind_textdomain_codeset = cc.has_function('bind_textdomain_codeset', dependencies: libintl_deps)
--else
--  libintl = subproject('proxy-libintl').get_variable('intl_dep')
--  libintl_deps = [libintl]
--  have_bind_textdomain_codeset = true  # proxy-libintl supports it
--endif
 +libintl = dependency('Intl', method:'cmake', required : true)
 +libintl_deps = [libintl]
-+have_bind_textdomain_codeset = cc.has_function('bind_textdomain_codeset',
-+                                                   dependencies : libintl_deps)
- 
- glib_conf.set('HAVE_BIND_TEXTDOMAIN_CODESET', have_bind_textdomain_codeset)
- 
-
++if false
+   # libintl supports different threading APIs, which may not
+   # require additional flags, but it defaults to using pthreads if
+   # found. Meson's "threads" dependency does not allow you to

--- a/ports/glib/portfile.cmake
+++ b/ports/glib/portfile.cmake
@@ -1,23 +1,23 @@
-set(GLIB_MAJOR_MINOR 2.74)
-set(GLIB_PATCH 0)
+vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
 
+string(REGEX MATCH "^([0-9]*[.][0-9]*)" GLIB_MAJOR_MINOR "${VERSION}")
 vcpkg_download_distfile(GLIB_ARCHIVE
-    URLS "https://download.gnome.org/sources/glib/${GLIB_MAJOR_MINOR}/glib-${GLIB_MAJOR_MINOR}.${GLIB_PATCH}.tar.xz"
-    FILENAME "glib-${GLIB_MAJOR_MINOR}.${GLIB_PATCH}.tar.xz"
-    SHA512 5cdadd2f4568c0c3d45083b4d39699abf651e42e020f7bc880cce3ff33d28943118388d17a0632777e843f48009c1f97d5634fde3cb8c69c7c7f35b278ac8225
+    URLS "https://download.gnome.org/sources/glib/${GLIB_MAJOR_MINOR}/glib-${VERSION}.tar.xz"
+    FILENAME "glib-${VERSION}.tar.xz"
+    SHA512 21176cb95fcab49a781d02789bf21191a96a34a6391f066699b3c20b414b3169c958bd86623deb34ca55912083862885f7a7d12b67cc041467da2ba94d9e83c3
 )
 
-vcpkg_extract_source_archive(
-    SOURCE_PATH
-    ARCHIVE ${GLIB_ARCHIVE}
+vcpkg_extract_source_archive(SOURCE_PATH
+    ARCHIVE "${GLIB_ARCHIVE}"
     PATCHES
         use-libiconv-on-windows.patch
         libintl.patch
 )
 
+vcpkg_list(SET OPTIONS)
 if (selinux IN_LIST FEATURES)
-    if(NOT VCPKG_TARGET_IS_WINDOWS AND NOT EXISTS "/usr/include/selinux")
-        message("Selinux was not found in its typical system location. Your build may fail. You can install Selinux with \"apt-get install selinux libselinux1-dev\".")
+    if(NOT EXISTS "/usr/include/selinux")
+        message(WARNING "SELinux was not found in its typical system location. Your build may fail. You can install SELinux with \"apt-get install selinux libselinux1-dev\".")
     endif()
     list(APPEND OPTIONS -Dselinux=enabled)
 else()
@@ -37,75 +37,64 @@ endif()
 vcpkg_configure_meson(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        -Dinstalled_tests=false
         ${OPTIONS}
+        -Dgtk_doc=false
+        -Dinstalled_tests=false
+        -Dlibelf=disabled
+        -Dman=false
         -Dtests=false
         -Dxattr=false
-        -Dlibelf=disabled
 )
-
 vcpkg_install_meson(ADD_BIN_TO_PATH)
-
 vcpkg_copy_pdbs()
 
-set(GLIB_TOOLS gdbus
-               gio
-               gio-querymodules
-               glib-compile-resources
-               glib-compile-schemas
-               gobject-query
-               gresource
-               gsettings
-               )
-
-if(NOT VCPKG_TARGET_IS_WINDOWS)
-    if(NOT VCPKG_TARGET_IS_OSX)
-        list(APPEND GLIB_TOOLS gapplication)
-    endif()
-    list(APPEND GLIB_TOOLS glib-gettextize gtester)
+file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
+set(GLIB_SCRIPTS
+    gdbus-codegen
+    glib-genmarshal
+    glib-gettextize
+    glib-mkenums
+    gtester-report
+)
+if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
+    list(REMOVE_ITEM GLIB_SCRIPTS glib-gettextize)
 endif()
-set(GLIB_SCRIPTS gdbus-codegen glib-genmarshal glib-mkenums gtester-report)
-
-
-if(VCPKG_TARGET_IS_WINDOWS AND VCPKG_TARGET_ARCHITECTURE MATCHES "x64|arm64")
-    list(APPEND GLIB_TOOLS  gspawn-win64-helper${VCPKG_EXECUTABLE_SUFFIX}
-                            gspawn-win64-helper-console${VCPKG_EXECUTABLE_SUFFIX})
-elseif(VCPKG_TARGET_IS_WINDOWS AND VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
-    list(APPEND GLIB_TOOLS  gspawn-win32-helper${VCPKG_EXECUTABLE_SUFFIX}
-                            gspawn-win32-helper-console${VCPKG_EXECUTABLE_SUFFIX})
-endif()
-vcpkg_copy_tools(TOOL_NAMES ${GLIB_TOOLS} AUTO_CLEAN)
 foreach(script IN LISTS GLIB_SCRIPTS)
     file(RENAME "${CURRENT_PACKAGES_DIR}/bin/${script}" "${CURRENT_PACKAGES_DIR}/tools/${PORT}/${script}")
     file(REMOVE "${CURRENT_PACKAGES_DIR}/debug/bin/${script}")
 endforeach()
 
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" OR NOT VCPKG_TARGET_IS_WINDOWS)
-    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+set(GLIB_TOOLS
+    gapplication
+    gdbus
+    gio
+    gio-querymodules
+    glib-compile-resources
+    glib-compile-schemas
+    gobject-query
+    gresource
+    gsettings
+    gtester
+)
+if(VCPKG_TARGET_IS_WINDOWS)
+    list(REMOVE_ITEM GLIB_TOOLS gapplication gtester)
+    if(VCPKG_TARGET_ARCHITECTURE MATCHES "x64|arm64")
+        list(APPEND GLIB_TOOLS gspawn-win64-helper gspawn-win64-helper-console)
+    elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+        list(APPEND GLIB_TOOLS gspawn-win32-helper gspawn-win32-helper-console)
+    endif()
+elseif(VCPKG_TARGET_IS_OSX)
+    list(REMOVE_ITEM GLIB_TOOLS gapplication)
 endif()
+vcpkg_copy_tools(TOOL_NAMES ${GLIB_TOOLS} AUTO_CLEAN)
 
-IF(VCPKG_TARGET_IS_WINDOWS)
-    set(SYSTEM_LIBRARIES dnsapi iphlpapi winmm lshlwapi)
-else()
-    set(SYSTEM_LIBRARIES resolv mount blkid selinux)
-endif()
-if(EXISTS "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/gio-2.0.pc")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/gio-2.0.pc" "\${bindir}" "\${prefix}/tools/${PORT}")
-endif()
-if(EXISTS "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/gio-2.0.pc")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/gio-2.0.pc" "\${bindir}" "\${prefix}/tools/${PORT}")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/glib-2.0.pc" "\${bindir}" "\${prefix}/tools/${PORT}")
+if(NOT VCPKG_BUILD_TYPE)
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/gio-2.0.pc" "\${bindir}" "\${prefix}/../tools/${PORT}")
-endif()
-if(EXISTS "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/glib-2.0.pc")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/glib-2.0.pc" "\${bindir}" "\${prefix}/tools/${PORT}")
-endif()
-if(EXISTS "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/glib-2.0.pc")
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/glib-2.0.pc" "\${bindir}" "\${prefix}/../tools/${PORT}")
 endif()
-vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES ${SYSTEM_LIBRARIES})
-
-file(INSTALL "${SOURCE_PATH}/LICENSES/LGPL-2.1-or-later.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_fixup_pkgconfig()
 
 # Fix python scripts
 set(_file "${CURRENT_PACKAGES_DIR}/tools/${PORT}/gdbus-codegen")
@@ -114,10 +103,16 @@ string(REPLACE "elif os.path.basename(filedir) == 'bin':" "elif os.path.basename
 string(REPLACE "path = os.path.join(filedir, '..', 'share', 'glib-2.0')" "path = os.path.join(filedir, '../..', 'share', 'glib-2.0')" _contents "${_contents}")
 string(REPLACE "path = os.path.join(filedir, '..')" "path = os.path.join(filedir, '../../share/glib-2.0')" _contents "${_contents}")
 string(REPLACE "path = os.path.join('${CURRENT_PACKAGES_DIR}/share', 'glib-2.0')" "path = os.path.join('unuseable/share', 'glib-2.0')" _contents "${_contents}")
-
 file(WRITE "${_file}" "${_contents}")
 
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/share/gdb")
-if(EXISTS "${CURRENT_PACKAGES_DIR}/tools/glib/glib-gettextize")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/glib/glib-gettextize" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/../..")
+if(EXISTS "${CURRENT_PACKAGES_DIR}/tools/${PORT}/glib-gettextize")
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/${PORT}/glib-gettextize" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/../..")
 endif()
+
+file(REMOVE_RECURSE
+    "${CURRENT_PACKAGES_DIR}/debug/include"
+    "${CURRENT_PACKAGES_DIR}/debug/share"
+    "${CURRENT_PACKAGES_DIR}/share/gdb"
+)
+
+file(INSTALL "${SOURCE_PATH}/LICENSES/LGPL-2.1-or-later.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/glib/vcpkg.json
+++ b/ports/glib/vcpkg.json
@@ -1,9 +1,9 @@
 {
   "name": "glib",
-  "version": "2.74.0",
+  "version": "2.74.1",
   "description": "Portable, general-purpose utility library.",
   "homepage": "https://developer.gnome.org/glib/",
-  "license": "LGPL-2.1-only",
+  "license": "LGPL-2.1-or-later",
   "supports": "!uwp",
   "dependencies": [
     "dirent",
@@ -26,7 +26,8 @@
       ]
     },
     "selinux": {
-      "description": "Build with selinux support."
+      "description": "Build with selinux support.",
+      "supports": "linux"
     }
   }
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2649,7 +2649,7 @@
       "port-version": 0
     },
     "glib": {
-      "baseline": "2.74.0",
+      "baseline": "2.74.1",
       "port-version": 0
     },
     "glibmm": {

--- a/versions/g-/glib.json
+++ b/versions/g-/glib.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "62e6236281de8dd6bd983d4d51721374e5559c7c",
+      "version": "2.74.1",
+      "port-version": 0
+    },
+    {
       "git-tree": "97fc41e084d04073610421cbdf4bfae0e89fb99c",
       "version": "2.74.0",
       "port-version": 0

